js/pages/estimation/components/cca-executions.js

```diff
@@ -0,0 +1,208 @@
+						return `${Math.floor(durationSec / 60)} min ${Math.round(durationSec % 60)} sec`;
+					}
+				},
+				{
+					title: 'Results',
+					className: this.classes('col-exec-results'),
+					render: (s, p, d) => {
+						return (d.status === this.estimationStatusGenerationOptions.COMPLETED || d.status === this.estimationStatusGenerationOptions.FAILED) && this.isResultsViewPermitted(d.id) && d.numResultFiles > 0 ?
+							`<a href='#' data-bind="css: $component.classes('reports-link'), click: $component.downloadResults.bind(null, id)"><i class="estimation-generation__action-ico fa fa-download"></i>Download ${d.numResultFiles} files</a>` : '-';
+					}
+				},
+			];
+			this.executionGroups = ko.observableArray([]);
+			if (this.isViewGenerationsPermitted()) {
+				this.loadData();
+				this.intervalId = PollService.add(() => {
+					this.loadData({ silently: true });
+				}, 10000);
+			}
+		}
+
+		dispose() {
+			PollService.stop(this.intervalId);
+		}
+
+		isGeneratePermitted(sourceKey) {
+			return !this.dirtyFlag().isDirty() && PermissionService.isPermittedGenerate(sourceKey, this.analysisId()) && config.api.isExecutionEngineAvailable();
+		}
+
+		isResultsViewPermitted(id) {
+			return PermissionService.isPermittedViewResults(id);
+		}
+
+		isViewGenerationsPermittedResolver() {
+			return ko.computed(() => this.analysisId() ? PermissionService.isPermittedListGenerations(this.analysisId()) : true);
+		}
+
+		loadData({silently = false} = {}) {
+			!silently && this.loading(true);
+			Promise.all([
+				SourceService.loadSourceList(),
+				EstimationService.listGenerations(this.analysisId()),
+			]).then(([
+								 allSources,
+								 executionList,
+							 ]) => {
+				let sourceList = allSources.filter(source => {
+					return (source.daimons.filter(function (daimon) { return daimon.daimonType === "CDM"; }).length > 0
+						&& source.daimons.filter(function (daimon) { return daimon.daimonType === "Results"; }).length > 0);
+				});
+
+				sourceList = lodash.sortBy(sourceList, ["sourceName"]);
+
+				sourceList.forEach(s => {
+					let group = this.executionGroups().find(g => g.sourceKey === s.sourceKey);
+					if (!group) {
+						group = {
+							sourceKey: s.sourceKey,
+							sourceName: s.sourceName,
+							submissions: ko.observableArray(),
+							status: ko.observable(),
+						};
+						this.executionGroups.push(group);
+					}
+
+					group.submissions((executionList && executionList.filter(e => e.sourceKey === s.sourceKey)) || []);
+					group.status(group.submissions() && group.submissions().find(s => s.status === this.estimationStatusGenerationOptions.STARTED) ?
+						this.estimationStatusGenerationOptions.STARTED :
+						this.estimationStatusGenerationOptions.COMPLETED);
+				});
+
+			}).finally(() => this.loading(false));
+		}
+
+		generate(sourceKey) {
+
+			let confirmPromise;
+
+			const executionGroup = this.executionGroups().find(g => g.sourceKey === sourceKey);
+			if (!executionGroup) {
```

Comment:
Such code of block (checking the parallel execution) already exists in Cohort Characterization: cannot we move it to shared utils?